File: iosApp/iosApp/ViewModel/UserProfileViewModel.swift
//
//  UserProfileViewModel.swift
//  iosApp
//
//  Created by Chinthaka on 05/04/2021.
//  Copyright © 2021 orgName. All rights reserved.
//

import Foundation
import shared


class UserProfileViewModel {
    private let sharedViewModel: SharedViewModel = SharedViewModel()
    
    public func editUserProfile(firstName : String,lastName:String ,completion: @escaping (UserProfileScreenState)->()) {
        let userProfile = UserProfile(firstName: firstName,lastName: lastName)
        self.sharedViewModel.stateProvider.getUserProfileState(user: userProfile) { (state : UserProfileScreenState) in
            completion(state)
        }
    }
    
    func initProfile(completion: @escaping (UserProfileScreenState)->()){
        self.sharedViewModel.stateProvider.doInitProfileScreen { (state: UserProfileScreenState) in
            completion(state)
        }
    }
}

File: androidApp/src/main/java/com/dev/kmmwallet/androidApp/viewmodel/UserProfileViewModel.kt
package com.dev.kmmwallet.androidApp.viewmodel

import com.dev.kmmwallet.shared.datasource.webservice.model.request.UserProfile
import com.dev.kmmwallet.shared.viewmodel.event.userprofile.UserProfileScreenState
import com.dev.kmmwallet.shared.viewmodel.event.userprofile.getUserProfileState
import com.dev.kmmwallet.shared.viewmodel.event.userprofile.initProfileScreen

class UserProfileViewModel : BaseViewModel() {

    fun updateUserProfile(
        firstName: String,
        lastName: String,
        updateUi: (profileState: UserProfileScreenState) -> Unit
    ) {
        sharedViewModel.stateProvider.getUserProfileState(
            UserProfile(firstName, lastName),
            fun(profileState) {
                updateUi(profileState)
            })
    }

    fun initProfile(updateProfileUi: (profileScreenState: UserProfileScreenState) -> Unit) {
        sharedViewModel.stateProvider.initProfileScreen {
            updateProfileUi(it)
        }
    }
}